python/graphstorm/config/argument.py

```diff
@@ -1176,6 +1176,50 @@ def lp_decoder_type(self):
         # Set default value to distmult
         return BUILTIN_LP_DISTMULT_DECODER
 
+    @property
+    def lp_edge_weight_for_loss(self):
+        """ The edge data fields that stores the edge weights used
+            in computing link prediction loss
+
+            The edge_weight can be in following format:
+            1) [weight_name]: global weight name, if an edge has weight,
+            the corresponding weight name is <weight_name>
+            2) ["src0,rel0,dst0:weight0","src0,rel0,dst0:weight1",...]:
+            different edge types have different edge weights.
+
+            By default, it is none.
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_lp_edge_weight_for_loss"):
+            assert self.task_type == BUILTIN_TASK_LINK_PREDICTION, \
+                "Edge weight for loss only works with link prediction"
+            edge_weights = self._lp_edge_weight_for_loss
+            if len(edge_weights) == 1 and \
+                ":" not in edge_weights[0]:
+                # global feat_name
+                return edge_weights[0]
+
+            # per edge type feature
+            weight_dict = {}
+            for weight_name in edge_weights:
+                weight_info = weight_name.split(":")
+                etype = tuple(weight_info[0].split(","))
+                if etype in weight_dict:
+                    assert False, \
+                        f"You already specify the weight names of {etype}" \
+                        f"as {weight_dict[etype]}"
+
+                # TODO: if train_etype is None, we need to check if
+                # etype exists in g.
+                assert self.train_etype is None or etype in self.train_etype, \
+                    f"{etype} must in the training edge type list"
+                assert isinstance(weight_info[1], str), \
+                    f"Feature name of {etype} should be a string not {weight_info[1]}"
+                weight_dict[etype] = [weight_info[1]]
```

Comment:
why does the value have to be a list?

Reply:
For extracting edge features from graph, I implemented a general function `prepare_batch_edge_input` which can be used by edge_weight_for_loss and edge_feature_for_decoder.
To follow `prepare_batch_edge_input`'s implementation, we return a list here.

Reply:
i think what is inconsistent is that line 1221 returns a string, but here it returns a list of one string for each edge type.

Reply:
It follows the same return type as L426 node_feat_name.
Returning a single string allows users to use one string to specify edge weights, whiling returning a dict of list of one string allows users to use different name to specify edge weights for different edge types.

Do you think we can only keep the `single string` option and force our user to use the same feature name for edge weights?

Reply:
No. One edge weight tensor for an edge type. 
But in order to make `prepare_batch_edge_input` more generate, We use dict of list of strings instead of dict of strings here.